GitCommands/Git/GitModule.cs

```diff
@@ -1612,15 +1612,11 @@ private string GetFetchArgs(string remote, string remoteBranch, string localBran
                 remoteBranchArguments = "+" + FormatBranchName(remoteBranch);
             }
 
-            string localBranchArguments;
+            string localBranchArguments = "";
             var remoteUrl = GetSetting(string.Format(SettingKeyString.RemoteUrl, remote));
 
             if (PathIsUrl(remote) && !string.IsNullOrEmpty(localBranch) && string.IsNullOrEmpty(remoteUrl))
                 localBranchArguments = ":" + GitCommandHelpers.GetFullBranchName(localBranch);
-            else if (string.IsNullOrEmpty(localBranch) || PathIsUrl(remote) || string.IsNullOrEmpty(remoteUrl))
-                localBranchArguments = "";
-            else
-                localBranchArguments = ":" + "refs/remotes/" + remote.Trim() + "/" + localBranch;
```

Comment:
Along with this change, the corresponding paths on the GUI side should be disabled. Otherwise an user could have a false impression that filling out the local branch field takes some effect.

Reply:
But it does take an effect, only if the remote is a URL. Do you want to enable/disable the local branch field according to the remote?

Reply:
>Do you want to enable/disable the local branch field according to the remote?

This could be a solution. Another one could be to fetch the remoteBranch directly into the localBranch the same way as it is done in the 1619 line. I don't know - I have never used fetching into local branch. Perhaps there are users which use this feature. Let's wait a few days to give them an opportunity to comment on this.